Framework/API/src/FileFinder.cpp

```diff
@@ -559,6 +560,9 @@ std::vector<std::string> FileFinderImpl::findRuns(const std::string &hintstr, co
       throw std::invalid_argument("Malformed range of runs: " + *h);
     } else if ((range.count() == 2) && (!fileSuspected)) {
       std::pair<std::string, std::string> p1 = toInstrumentAndNumber(range[0]);
+      if (boost::algorithm::istarts_with(hint, "PG3")) {
```

Comment:
The method `toInstrumentAndNumber` currently contains some logic that has a special case for both `PG3` at SNS & `SANS2D` at ISIS: https://github.com/mantidproject/mantid/blob/391d05541e68d9d75e79b8b6ade4b306bd836483/Framework/API/src/FileFinder.cpp#L188

Can we adapt this in someway or (is it broken?) rather than adding further special cases? I assume I'm probably missing something in what is currently failing?

Reply:
There is a bunch of oddities with `FileFinder`:
1. Pervasive assumption that runnumbers are not in instrument names
2. Once the hint is split into `InstrumentName` and runnumber, the `InstrumentName` is converted back into a string before being combined with the runnumber to create a *new* hint that is parsed.
3. Even though the hint can suggest multiple runnumbers, the return only supports returning a single filename (unless there is a extra bit that separates paths with a delimiter that the client splits on. The data catalog we are using supports searching for multiple files in a single rest call. If `FileFinder` could search for multiple files at once, we'd see an immediate performance improvement.

My intent is to let this one in (for now) and put together a design for some `FileFinder` improvements (aimed at items 1-2). I like the idea of a fuller rewrite (to take care of 3), but don't think it is practical at the moment.

Reply:
That's fair. Agreed that allowing multiple file searching in the archive finders would be helpful. It's possible we might start needing to look into these things as we're getting reports that file searching on our DA platform is slow but no commitment at the moment.